test/camunda-cloud/ElementTemplatesReplaceProviderSpec.js

```diff
@@ -0,0 +1,313 @@
+import {
+  inject,
+  getBpmnJS,
+  bootstrapCamundaCloudModeler
+} from 'test/TestHelper';
+
+import {
+  attr as domAttr,
+  queryAll as domQueryAll,
+  query as domQuery
+} from 'min-dom';
+
+import {
+  isTemplateApplied
+} from 'lib/camunda-cloud/features/replace/ElementTemplatesReplaceProvider';
+
+import simpleXml from '../fixtures/replaceElementTemplates.bpmn';
+import templates from './replace-element-templates.json';
+
+
+describe('<ElementTemplatesReplaceProvider>', function() {
+
+  beforeEach(bootstrapCamundaCloudModeler(
+    simpleXml
+  ));
+
+  beforeEach(inject(function(elementTemplates) {
+    elementTemplates.set(templates);
+  }));
+
+
+  describe('display', function() {
+
+    it('should display matching element templates', inject(function(elementRegistry) {
+
+      // given
+      const task = elementRegistry.get('ServiceTask_1');
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(5);
+
+    }));
+
+
+    it('should not display element templates that do not apply', inject(function(elementRegistry) {
+
+      // given
+      const task = elementRegistry.get('Task_1');
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(4);
+
+    }));
+
+
+    it('should not display applied element template', inject(function(elementRegistry, elementTemplates) {
+
+      // given
+      const task = elementRegistry.get('ServiceTask_1');
+      elementTemplates.applyTemplate(task, templates[0]);
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(4);
+
+    }));
+
+
+    it('display options to reset to plain element in correct order', inject(function(elementRegistry, elementTemplates) {
```

Comment:
What about the glitch you found @smbea. Shall we test cover it, too?

Reply:
Exactly. If it is a quick one for you add a failing test case (`it.skip`) and we can very conveniently fix that issue in the future.

Reply:
Thanks to your test case the fix was straight forward: https://github.com/camunda/camunda-bpmn-js/pull/207/commits/96abacd15dd5ec476c3f8bde48439b3642fc5ffb

IN the past we did just concatenate all option groups; to determine a proper position we have to search _WITHIN_ a group, not the whole list.